spmd/tensor/ops/tensor_ops.py

```diff
@@ -53,3 +54,36 @@ def no_shard_prop_rule(op_schema: OpSchema) -> OutputSharding:
 
 for op in no_shard_prop_ops:
     DTensor._op_to_rules[op] = no_shard_prop_rule
+
+
+@register_prop_rule("aten.bucketize.Tensor")
+def prop_bucketize(op_schema: OpSchema) -> OutputSharding:
+    """
+    Point-wise on the first input (just propagate input sharding).
+    Expect replicated for second input.
+    """
+    input_schema, boundaries = op_schema.args_schema
+    assert isinstance(input_schema, DTensorSpec)
+    assert isinstance(boundaries, DTensorSpec)
+
+    if all(isinstance(p, Replicate) for p in boundaries.placements):
+        return OutputSharding(output_spec=input_schema)
+    else:
+        return OutputSharding(
```

Comment:
Are we suggesting Replicate() for the second input if either of the inputs are Shard? Can you add a comment here as well?